syntax = "proto3";
package DataCenter;

// DataCenter function is start by 201-300
enum Functions{
    Invalid=0;

    // Connect to Database. It will return the connect result.
    // Request:ConnectDatabaseRequest
    // Response:ConnectDatabaseResponse
    ConnectDatabase=1;

    // Run query sql.
    // Request: QuerySQLCmdRequest
    // Response: QuerySQLCmdResponse
    QuerySQLCmd=2;

    // Run insert sql
    // Request: InsertSQLCmdRequest
    // Response: InsertSQLCmdResponse
    InsertSQLCmd=3;

    // Run update sql
    // Request: UpdateSQLCmdRequest
    // Response: UpdateSQLCmdResponse
    UpdateSQLCmd=4;

    // Run delete sql
    // Request: DeleteSQLCmdRequest
    // Response: DeleteSQLCmdResponse
    DeleteSQLCmd=5;

}

message ConnectDatabaseRequest{
    string Database=1;
}

message ConnectDatabaseResponse{
    int32 Session=1;
}

message QuerySQLCmdRequest{
    int32 Session=1;
    string SQLCMD=2;
}

message SQLResultLineItem{
    repeated string RawDataItem=1;
}

message QuerySQLCmdResponse{
    repeated SQLResultLineItem Lines=2;
}

message InsertSQLCmdRequest{
    int32 Session=1;
    string SQLCMD=2;
    // If set true, will return "SELECT @@IDENTITY" value.
    bool GetIdentity=3;
}

message InsertSQLCmdResponse{
    int32 AffectedRows=1;
    // If no set GetIdentity, return 0. else return the value. if get fialed, return -1;
    int32 Identity=2;
}

message UpdateSQLCmdRequest{
    int32 Session=1;
    string SQLCMD=2;
}

message UpdateSQLCmdResponse{
    int32 AffectedRows=1;
}

message DeleteSQLCmdRequest{
    int32 Session=1;
    string SQLCMD=2;
}

message DeleteSQLCmdResponse{
    int32 AffectedRows=1;
}